controllers/v1/observationsController.js

```diff
@@ -982,42 +1064,41 @@ module.exports = class Observations extends Abstract {
                     result: {}
                 };
 
-                let observationDocument = await database.models.observations.findOne({ 
-                    _id: req.params._id, 
-                     createdBy: req.userDetails.userId,
-                     status: {$ne:"inactive"}, 
-                     entities: ObjectId(req.query.entityId) }).lean();
+                let observationDocument = await observationsHelper.findObservation(
+                    req.params._id, 
+                    req.userDetails.userId, 
+                    req.query.entityId
+                )
 
                 if (!observationDocument) {
                     return resolve({ 
                         status: httpStatusCode.bad_request.status, 
                         message: messageConstants.apiResponses.OBSERVATION_NOT_FOUND
                     });
                 }
-               
-                let entityQueryObject = { _id: req.query.entityId, entityType: observationDocument.entityType };
-                let entityDocument = await database.models.entities.findOne(
-                    entityQueryObject,
-                    {
-                        metaInformation: 1,
-                        entityTypeId: 1,
-                        entityType: 1,
-                        registryDetails: 1
-                    }
-                ).lean();
+              
+                let filterData = {
+                    "id" : req.query.entityId,
+                    "type" : observationDocument.entityType
+                };
 
-                if (!entityDocument) {
-                    let responseMessage = messageConstants.apiResponses.ENTITY_NOT_FOUND;
+                let entitiesDocument = await sunbirdService.learnerLocationSearch( filterData );
+
+                if ( !entitiesDocument.success || !entitiesDocument.data || !entitiesDocument.data.response.length > 0 ) {
```

Comment:
@VISHNUDAS-tunerlabs entitiesDocument.data.response will always present in response?